cmd2/utils.py

```diff
@@ -281,10 +281,7 @@ def write(self, b: bytes) -> None:
                 raise TypeError('a bytes-like object is required, not {}'.format(type(b)))
             self.byte_buf += b
             if self.echo:
-                if hasattr(self.inner_stream, 'buffer'):
-                    self.inner_stream.buffer.write(b)
-                else:
-                    self.inner_stream.write(b.decode(encoding=self.encoding, errors=self.errors))
+                self.inner_stream.buffer.write(b)
```

Comment:
I don't think this is a 100% safe change.  I vaguely recall when manually testing this class in a variety of ways that there are some small minority of things that could be used as an inner_stream which do **NOT** have a buffer attribute. 

Thus, I would recommend leaving this code block as is.

Reply:
Actually now that you refactored the part that was calling this, I think this is ok